tradebot/utils/rate_limiter.py

```
@@ -0,0 +1,389 @@
+                        requests_used INTEGER NOT NULL DEFAULT 0,
+                        last_request_time REAL NOT NULL DEFAULT 0.0,
+                        created_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP
+                    )
+                ''')
+                conn.commit()
+            
+            logger.info(f"Quota database initialized: {self.storage_path}")
+            
+        except sqlite3.Error as e:
+            logger.error(f"Failed to initialize quota database: {e}")
+            raise RuntimeError(f"Quota database initialization failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)